### DISCUSSION TITLE
What should we do with naming and awkward URLs

### DISCUSSION THREAD

**Discussion by GingerSnap-xx** [mampfes/hacs_waste_collection_schedule#1898] [Q&A] [answered]
Some of the documentation mentions naming things like your branch or your integration after the URL where you will be getting your schedule from. The issue I'm facing is that the host / url of the municipality is different than their trash schedule. They are using an arcgis map tool to show you collection information. I don't want to name it "arcgis.com" but I guess "planogis.maps.arcgis.com" would avoid ambiguity, but that's still not the host of the municipality itself. Which should I go with or something else? Is the most value placed on having the _municipalities_ properly linked to, or the literal resource serving the data?

**Comment by 5ila5** [comment, marked as answer]
Does this source support multiple municipalities, or do you need to make changes for every municipality?

If you can realize the source for multiple municipalities, I would name the source arcgis_com and add the supported municipalities with their URLs to the EXTRA_INFO section. ([example](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/custom_components/waste_collection_schedule/waste_collection_schedule/source/environmentfirst_co_uk.py)). The update_docu_links.py will then add arcgis.com and the municipalities as extra elements in the README.md.

If every there is no easy way to support multiple municipalities with one source, I would choose the municipalities' website for naming as it prevents naming conflicts, and it's more clear for the user.

**Reply by GingerSnap-xx** [reply]
It looks like this municipality builds upon arcgis in a custom way so it doesn't seem like it could easily support multiple municipalities. I'm fighting trying to understand the API so it will be a while. I'll start with naming it after the site for the municipality and if I come to feel like it is a generic solution for anyone using some standard tool built into or upon arcgis, then I'll adjust accordingly. Thanks!
